src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1591 @@
+function plugindef()
+    -- This function and the 'finaleplugin' namespace
+    -- are both reserved for the plug-in definition.
+    finaleplugin.RequireSelection = true
+    finaleplugin.Author = "Jacob Winkler"
+    finaleplugin.Copyright = "2022"
+    finaleplugin.Version = "2.0"
+    finaleplugin.Date = "2022-07-06"
+    return "Harp Pedal Wizard", "Harp Pedal Wizard", "Creates Harp Diagrams and Pedal Changes"
+end
+
+if finenv.IsRGPLua == false then
+    local ui = finenv.FCUI
+    ui:AlertInfo("This script requires RGP Lua to function.", NULL)
+    local str = finale.FCString()
+    str.LuaString = "https://robertgpatterson.com/-fininfo/-rgplua/rgplua.html"
+    ui:DisplayWebURL(str)
+end
+
+local library = require("library.general_library")
+--local configuration = require("library.configuration")
+
+local config_filename = "com.harp_pedal_wizard.text"
+
+-- finalelua library functions - copied from library\configuration.lua
+local strip_leading_trailing_whitespace = function (str)
+    return str:match("^%s*(.-)%s*$") -- lua pattern magic taken from the Internet
+end
+
+local parse_parameter -- forward function declaration
+
```

Comment:
As as a matter of interest, Lua does not require a forward function declaration. The only thing that is required is that `parse_parameter` be defined as a function by the first time you try to call it in `parse_table`. This "forward declaration" actually does nothing.